[STREAMPIPES-343] Added timeinterval between the two events

Samarth08 opened the pull request:
<!--
  ~ Licensed to the Apache Software Foundation (ASF) under one or more
  ~ contributor license agreements.  See the NOTICE file distributed with
  ~ this work for additional information regarding copyright ownership.
  ~ The ASF licenses this file to You under the Apache License, Version 2.0
  ~ (the "License"); you may not use this file except in compliance with
  ~ the License.  You may obtain a copy of the License at
  ~
  ~    http://www.apache.org/licenses/LICENSE-2.0
  ~
  ~ Unless required by applicable law or agreed to in writing, software
  ~ distributed under the License is distributed on an "AS IS" BASIS,
  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  ~ See the License for the specific language governing permissions and
  ~ limitations under the License.
  ~
  -->
  
Fixes: https://issues.apache.org/jira/browse/STREAMPIPES-343

bossenti commented:
@Samarth08, thanks for your proposal.

Unfortunately, your solution does not yet produce to the desired behavior.
The FileSetAdapter is meant to read a file, e.g. csv files, json files etc., at once and create an event for each instance described by the file.
In contrast to processing elements, it does not have an `onEvent` method.
Instead, the `run` method is called once when the adapter is invoked (in this case, when the corresponding pipeline is started).
Accordingly, the `parser.parse(in, stk)` emits all instances in the file before the waiting time begins and the `run` method ends.

Therefore, we need to come up with something different here. For example, we could extend the implementation of the `parse` method or create something similar to the `ZipFileIterator`  you already imported.

Samarth08 commented:
@bossenti I have added parse method which will emit individual events in the file.I have added time interval between each event.Let me know if this works for you.

bossenti commented on streampipes-connect-adapters/src/main/java/org/apache/streampipes/connect/protocol/set/FileProtocol.java:
```diff
@@ -79,11 +84,32 @@ public ProtocolDescription declareModel() {
     public Protocol getInstance(ProtocolDescription protocolDescription, Parser parser, Format format) {
         StaticPropertyExtractor extractor = StaticPropertyExtractor.from(protocolDescription.getConfig());
-
+        this.timeBetweenReplay = extractor.singleValueParameter(INTERVAL_KEY, Integer.class);
```
```suggestion
        int timeBetweenReplay = extractor.singleValueParameter(INTERVAL_KEY, Integer.class);
```

bossenti commented on streampipes-connect-adapters/src/main/java/org/apache/streampipes/connect/protocol/set/FileProtocol.java:
```diff
@@ -79,11 +84,32 @@ public ProtocolDescription declareModel() {
-
+        this.timeBetweenReplay = extractor.singleValueParameter(INTERVAL_KEY, Integer.class);
         String fileFetchUrl = extractor.selectedFileFetchUrl("filePath");
         return new FileProtocol(parser, format, fileFetchUrl);
```
```suggestion
        return new FileProtocol(parser, format, fileFetchUrl, timeBetweenReplay);
```
Please adapt the constructor accordingly

bossenti reviewed (changes requested):
@Samarth08, thanks for your improvements.
With the requested changes the adapter works as expected

Samarth08 commented:
@bossenti I have made the suggested changes.

bossenti commented:
looks great, thank you!
